refactor: pull _replace_node common functionality out of Scheduler.finalize_multi_template_buffers

aorenste opened the pull request:
Pull replace_node function out of Scheduler.finalize_multi_template_buffers(). This is needed by the next PR (#163369). As part of this also pull the _replace_operation_buffer() up to top-level since it needed no self references.

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #163369
* __->__ #163368



cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov @coconutruben

aorenste committed:
refactor: move replace_operation_buffer to global scope

[ghstack-poisoned]

aorenste committed:
Update on "refactor: move replace_operation_buffer to global scope"




[ghstack-poisoned]

aorenste committed:
Update on "refactor: move replace_operation_buffer to global scope"




[ghstack-poisoned]

aorenste committed:
Update on "refactor: move replace_operation_buffer to global scope"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

aorenste committed:
Update on "refactor: move replace_operation_buffer to global scope"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

aorenste committed:
Update on "refactor: move replace_operation_buffer to global scope"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

PaulZhang12 reviewed (changes requested):
Actually I think this needs a bit more context.. we are moving fields from object level to class level?

aorenste committed:
Update on "refactor: pull _replace_node common functionality out of Scheduler.finalize_multi_template_buffers"

Pull replace_node function out of Scheduler.finalize_multi_template_buffers(). This is needed by the next PR (#163369). As part of this also pull the _replace_operation_buffer() up to top-level since it needed no self references.




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

aorenste committed:
Update on "refactor: pull _replace_node common functionality out of Scheduler.finalize_multi_template_buffers"

Pull replace_node function out of Scheduler.finalize_multi_template_buffers(). This is needed by the next PR (#163369). As part of this also pull the _replace_operation_buffer() up to top-level since it needed no self references.




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

PaulZhang12 reviewed (approved):
Right, I understand, but I guess my initial assumption was that we usually just annotate class level variables (static) and assign instance level at init time. This refactor seems fundamentally better though, stamped.
